### PR TITLE
fix qgsDoubleString printing -0

### PR DESCRIPTION
**Pull request by 3nids**
In latest Qt versions (5.11 maybe 5.10), `QString::number(-0.000000000708115, 'f', 0)` prints `-0` and not `0` anymore.
I have reported upstream and this is not seen as a bug.

> That's also the result from C's printf and Python's.
> IEEE 754 floating point is not mathematically perfect and it does have the concept of negative zero.

https://bugreports.qt.io/browse/QTBUG-71439

This method is using in comparing geometries (via WKT) and leads to failing test.
To my understanding, the precision acts as a rounding and there is not reason to print -0.

If you think differently, please tell me why :)

### CODE REVIEW

**Comment by m-kuhn** [comment]
I agree that we prefer not having -0, I'm not sure about the overhead of adding QRegExp, I guess having a simple `== QLatin1String( "-0" )` will be way faster

File: src/core/qgis.h [hunk 1]
```diff
@@ -239,9 +239,15 @@ CORE_EXPORT uint qHash( const QVariant &variant );
 inline QString qgsDoubleToString( double a, int precision = 17 )
 {
   if ( precision )
-    return QString::number( a, 'f', precision ).remove( QRegExp( "\\.?0+$" ) );
+  {
+    return QString::number( a, 'f', precision ).remove( QRegularExpression( "\\.?0+$" ) );
```
Comment (nyalldawson) [inline src/core/qgis.h]: I wonder if there's a performance win possible here by making this regex static

**Comment by 3nids** [comment]
I've made a quick simulation.
* String comparison (`=="-0"`) is 10 times faster than the regex.
* making the regex static is 2 to 3 times faster (depending if it has ending 0s)

will update the code accordingly

**Comment by nyalldawson** [comment]
> QString::number is not locale-aware, shouldn't you use QLocale().toString ?

I don't believe this method should be locale aware -- it's used to write settings to projects mainly. I'd proceed with extreme caution here.

**Comment by elpaso** [comment]
@nyalldawson I'm not sure, we need to check if that's not used for number GUI representation, and document that is not locale aware anyway.

**Comment by 3nids** [comment]
I merged it as is.
The discussion of local or not can be followed and fixed in a later step if needed.

**Comment by 3nids** [comment]
I merged it as is.
The discussion of local or not can be followed and fixed in a later step if needed.

**Pull request merged**